pkg/util/restoptions/helpers.go

```diff
@@ -51,10 +52,14 @@ func ApplyOptions(optsGetter Getter, store *registry.Store, oldIsNamespaced bool
 	if store.CreateStrategy == nil {
 		return fmt.Errorf("store for %s must have CreateStrategy set", store.QualifiedResource.String())
 	}
-
-	isNamespaced := store.CreateStrategy.NamespaceScoped()
-	if isNamespaced != oldIsNamespaced { // TODO(soltysh): oldIsNamespaced should be completely removed in #12541
-		return fmt.Errorf("CreateStrategy has %v for namespace scope but user specified %v as namespace scope", isNamespaced, oldIsNamespaced)
+	if store.ObjectNameFunc == nil {
```

Comment:
We'll have to cherry-pick that upstream PR, it won't make k8s 1.6, but I agree it should be done.